File: software/extensions/libs/babylon/src/lib/objects/clustertool/clustertool.js
import {BabylonObject} from "../../objects";


export class ClusterTool extends BabylonObject {




    buildObject() {
        return undefined;
    }

    highlight(state) {
        return undefined;
    }

    update(data) {
        // here we can update the clustertool
        const angle = data.arm_angle ? data.arm_angle : 0;

    }

    test_function(x) {
        console.log(x);
    }

    onMessage(message) {
        return undefined;
    }
}

File: software/extensions/libs/babylon/src/lib/objects/clustertool/clustertool.py
import dataclasses

from extensions.libs.babylon.src.babylon import BabylonObject


@dataclasses.dataclass
class ClusterToolData:
    arm_angle: float = 0


class ClusterTool_BabylonObject(BabylonObject):
    type = 'clustertool'

    def __init__(self, object_id: str, **kwargs):
        super().__init__(object_id, **kwargs)
        self.data = ClusterToolData()

    def set_arm_angle(self, angle: float):
        self.data.arm_angle = angle
        self.update()

    def test(self, value):
        self.function(
            function_name='test_function',
            x=value
        )

    def getConfig(self) -> dict:
        config = {

        }
        return config

    def getData(self) -> dict:
        data = {
            'arm_angle': self.data.arm_angle
        }
        return data
